apps/deploy-web/src/components/remote-deploy/RemoteRepositoryDeployManager.tsx

```diff
@@ -3,69 +3,82 @@ import { Control, UseFormSetValue } from "react-hook-form";
+import { useGitLabFetchAccessToken, useGitLabUserProfile } from "@src/queries/useGitlabQuery";
+import { BitbucketService } from "@src/services/remote-deploy/bitbucket-http.service";
+import { GitHubService } from "@src/services/remote-deploy/github-http.service";
+import { GitLabService } from "@src/services/remote-deploy/gitlab-http.service";
+import { EnvVarUpdater } from "@src/services/remote-deploy/remote-deployment-controller.service";
 import { tokens } from "@src/store/remoteDeployStore";
 import { SdlBuilderFormValuesType, ServiceType } from "@src/types";
+import { RouteStep } from "@src/types/route-steps.type";
+import { UrlService } from "@src/utils/urlUtils";
+import { useBitFetchAccessToken, useBitUserProfile } from "../../queries/useBitBucketQuery";
 import BitBucketManager from "./bitbucket/BitBucketManager";
 import RemoteBuildInstallConfig from "./deployment-configurations/RemoteBuildInstallConfig";
 import RemoteDeployEnvDropdown from "./deployment-configurations/RemoteDeployEnvDropdown";
 import GithubManager from "./github/GithubManager";
 import GitlabManager from "./gitlab/GitlabManager";
-import { handleLoginBit, useBitFetchAccessToken, useBitUserProfile } from "./remote-deploy-api-queries/bit-bucket-queries";
-import { handleLogin, handleReLogin, useFetchAccessToken, useUserProfile } from "./remote-deploy-api-queries/github-queries";
-import { handleGitLabLogin, useGitLabFetchAccessToken, useGitLabUserProfile } from "./remote-deploy-api-queries/gitlab-queries";
 import AccountDropDown from "./AccountDropdown";
 import CustomInput from "./BoxTextInput";
-import { appendEnv, protectedEnvironmentVariables } from "./helper-functions";
-
 const RemoteRepositoryDeployManager = ({
   setValue,
   services,
   control,
   deploymentName,
   setDeploymentName,
-  setIsRepoDataValidated
+  setIsRepoInputValid
 }: {
   setValue: UseFormSetValue<SdlBuilderFormValuesType>;
   services: ServiceType[];
   control: Control<SdlBuilderFormValuesType>;
   setDeploymentName: Dispatch<string>;
   deploymentName: string;
-  setIsRepoDataValidated?: Dispatch<boolean>;
+  setIsRepoInputValid?: Dispatch<boolean>;
 }) => {
   const [token, setToken] = useAtom(tokens);
-  const [selectedTab, setSelectedTab] = useState("git");
 
+  const [selectedTab, setSelectedTab] = useState("git");
+  const router = useRouter();
   const [hydrated, setHydrated] = useState(false);
+  const isRepoAndBranchPresent = (env: Array<{ key: string }>) =>
+    env.some(e => e.key === protectedEnvironmentVariables.REPO_URL) && env.some(e => e.key === protectedEnvironmentVariables.BRANCH_NAME);
+
+  const isValid = isRepoAndBranchPresent(services?.[0]?.env || []);
+  const isRepoUrlDefault = (env: ServiceType["env"]) => env?.some(e => e.key === protectedEnvironmentVariables.REPO_URL && e.value === DEFAULT_ENV_IN_YML);
+
+  const shouldResetValue = isRepoUrlDefault(services?.[0]?.env || []);
+
+  const envVarUpdater = new EnvVarUpdater(services);
```

Comment:
I'm not sure we need a class here. Or if we do it should use like this  `useMemo(() => new EnvVarUpdate(services), [services])`